raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -830,3 +835,213 @@ def run_test_automatic_dispute(raiden_network, deposit, token_addresses):
+        secret_registry_address,
+        deposit,
+        blockchain_type,
+        retry_timeout,
+):
+    """Simulate the case where:
+    - A sends B a transfer
+    - B sends A a transfer
+    - Secrets were never revealed
+    - B closes channel
+    - Wait for settle
+    - A crashes
+    - Wait for unlock from B
+    - Restart A
+    At this point, the current unlock logic will try to unlock
+    iff the node gains from unlocking. Which means that the node will try to unlock
+    either side. In the above scenario, each node will unlock its side.
+    This test makes sure that we do NOT invalidate A's unlock transaction based
+    on the ContractReceiveChannelBatchUnlock caused by B's unlock.
+    """
+    alice_app, bob_app = raiden_network
+    registry_address = alice_app.raiden.default_registry.address
+    token_address = token_addresses[0]
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        chain_state=views.state_from_app(alice_app),
+        payment_network_id=alice_app.raiden.default_registry.address,
+        token_address=token_address,
+    )
+
+    hold_event_handler = HoldOffChainSecretRequest()
+    bob_app.raiden.raiden_event_handler = hold_event_handler
+    alice_app.raiden.raiden_event_handler = hold_event_handler
+
+    token_network = views.get_token_network_by_identifier(
+        chain_state=views.state_from_app(alice_app),
+        token_network_id=token_network_identifier,
+    )
+
+    channel_identifier = get_channelstate(alice_app, bob_app, token_network_identifier).identifier
+
+    assert channel_identifier in token_network.partneraddresses_to_channelidentifiers[
+        bob_app.raiden.address
+    ]
+
+    alice_to_bob_amount = 10
+    identifier = 1
+
+    alice_transfer_secret = sha3(alice_app.raiden.address)
+    alice_transfer_secrethash = sha3(alice_transfer_secret)
+
+    bob_transfer_secret = sha3(bob_app.raiden.address)
+    bob_transfer_secrethash = sha3(bob_transfer_secret)
+
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=alice_transfer_secrethash,
+    )
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=bob_transfer_secrethash,
+    )
+
+    alice_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=bob_app.raiden.address,
+        identifier=identifier,
+        secret=alice_transfer_secret,
+    )
+
+    bob_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=alice_app.raiden.address,
+        identifier=identifier + 1,
+        secret=bob_transfer_secret,
+    )
+
+    wait_for_state_change(
+        raiden=alice_app.raiden,
+        item_type=ActionInitTarget,
+        attributes={},
+        retry_timeout=retry_timeout,
+    )
+
+    wait_for_state_change(
```

Comment:
this works, but the test can be simpler if you use the async result returned by `HoldRaidenEvent` instead. I claim it's simpler because we remove the hot loop and the relation with the secret reveal becomes explicit